File: python/magicien.py
from dataclasses import dataclass, field
from typing import Optional
from livre_magique import LivreMagique

@dataclass
class Magicien:
    nom: str
    mon_livre: Optional[LivreMagique] = field(default=None, init=False)

    def prendre_livre(self, nouveau_livre: LivreMagique):
        self.mon_livre = nouveau_livre

    def calculer_puissance_totale(self) -> int:
        base = 10
        if self.mon_livre:
            return base + self.mon_livre.points_de_magie
        return base

File: java/Magicien.java
public class Magicien {
    private String nom;
    private LivreMagique monLivre; // Association 0..1 

    public Magicien(String unNom) {
        this.nom = unNom;
        this.monLivre = null; // Mains vides au départ 
    }

    // Méthode pour relier les deux objets
    public void prendreLivre(LivreMagique livre) {
        this.monLivre = livre;
    }
    
    public int calculerPuissanceSort() {
        int baseMage = 10;
            if (this.monLivre != null) {
                // Collaboration : on récupère les points du livre 
                return baseMage + this.monLivre.getPointsDeMagie(); 
            }
        return baseMage;
    }   
}